lib/std/crypto/25519/field.zig

```diff
@@ -307,12 +307,14 @@ pub const Fe = struct {
     }
 
     pub fn pow2523(a: Fe) Fe {
-        var c = a;
-        var i: usize = 0;
-        while (i < 249) : (i += 1) {
-            c = c.sq().mul(a);
-        }
-        return c.sq().sq().mul(a);
+        var t0 = a.mul(a.sq());
```

Comment:
Is the code generation worse if you make each of these `const`?
Is there any better name we can use for the temporaries? (or do they come from some paper?)

Reply:
This is the addition chain:

```text
_10    = 2*1
_11    = 1 + _10
_1100  = _11 << 2
_1111  = _11 + _1100
_11110 = 2*_1111
_11111 = 1 + _11110
x10    = _11111 << 5 + _11111
x15    = x10 << 5 + _11111
x30    = x15 << 15 + x15
x60    = x30 << 30 + x30
x120   = x60 << 60 + x60
x240   = x120 << 120 + x120
x250   = x240 << 10 + x10
x250 << 2 + 1
```

There's nothing exciting in addition chain code, so variable reuse was made to keep the function short.